### PR TITLE
[Feature](bangc-ops):add new op roi_align_rotated

### PR DESCRIPTION
**Pull request by tudejiang79**
Thanks for your contribution and we appreciate it a lot. 

## 1. Motivation

add new op roi_align_rotated

## 2. Modification

1) mlu-ops/bangc-ops/kernels/roi_align_rotated/*
2) mlu-ops/bangc-ops/mlu_op.h
3) mlu-ops/bangc-ops/mlu_op_kernel.h
4) mlu-ops/bangc-ops/kernel_depends.toml
5) mlu-ops/bangc-ops/test/mlu_op_gtest/pb_gtest/mlu_op_test_proto/*
6) mlu-ops/bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_align_rotated_forward/*
7) mlu-ops/bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_align_rotated_backward/*
8) mlu-ops/docs/bangc-docs/user_guide/9_operators/*
9) mlu-ops/docs/bangc-docs/design_docs/roi_align_rotated/*
## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](../docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS Accuracy Acceptance Standard](../docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- [x] diff1, diff1 <= 3e-3
- [x] diff2, diff2 <= 3e-3

#### 3.1.2 Operator Scheme checklist

|     No.      |           Details            |      Check Results       |
|----------------|---------------------------|---------------------|
|        1       |          Supported hardware         | MLU270 <br> MLU290 <br>MLU370|
|        2       |          Job types          |    block <br> U1 <br> U4    |
|        3       |         Layouts            |  NHWC 、NCHW、ARRAY etc    |
|        4       |         Whether multi-dimensions are supported              |                |
|        5       |          Whether element zero is supported             |               |
|        6       |         Data type(half/float)       |         half / float etc           |
|        7      |        Whether there is size limit           |             |

#### 3.1.3 New Feature Test

If you have checked the following items, please tick the relevant box.

- [x] Data type test
- [x] Multidimensional tensor test
- [x] Layout test
- [x] Different size/integer remainder end segment/alignment misalignment test
- [x] Zero dimensional tensor test/zero element test
- [x] stability test
- [x] Multiple platform test
- [x] Gen_case module test
- [x] Nan/INF tests 
- [ ] Bug fix tests
- [x] For memory leak check details, see[GTest-User-Guide-zh](../docs/GTest-User-Guide-zh.md).
- [x] For code coverage check details, see: [GTest-User-Guide-zh](../docs/GTest-User-Guide-zh.md).
- [x] For I/O calculation efficiency check details, see: [MLU-OPS Performance Acceptance Standard](../docs/MLU-OPS-Performance-Acceptance-Standard.md).

#### 3.1.4 Parameter Check

When a new operator is submitted, the test points are given and the test results are stated.

| Test Point         | Acceptance Standard | Test Result (Error Message) |
| -------------- | -------- | -------------------- |
| Whether it conforms to the operator restriction | Normal error |                      |
| Whether illegal parameters are passed  | Normal error |                      |

### 3.2 Performance Test

See [MLU-OPS Performance Acceptance Standard](../docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

Platform ：MLU270

|Operator|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-----|----|----|----|----|----|------|-----|
|op_name|   |    |     |    |    |    |     |
|op_name|   |    |     |    |    |    |     |

Platform ：MLU290

|Operator|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-----|----|----|----|----|----|------|-----|
|op_name|   |    |     |    |    |    |     |
|op_name|   |    |     |    |    |    |     |

Platform：MLU370

|Operator|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-----|----|----|----|----|----|------|-----|
|op_name|   |    |     |    |    |    |     |
|op_name|   |    |     |    |    |    |     |

### 3.3 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.

### CODE REVIEW

File: bangc-ops/mlu_op.h [hunk 1]
```diff
@@ -2293,6 +2293,240 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ * @param[in] clockwise
+ *   A boolean value which determines whether the rotation of roi is clockwise.
+ * @param[out] output_desc
+ *   Descriptor of output, containing dimension and the layout of output.
+ * @param[out] output
+ *   Pointer to the MLU memory that stores the output tensor.
+ * @par Return
+ * - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM.
+ *
+ * @par Formula
```
Comment (wushaoqiang2168) [inline bangc-ops/mlu_op.h:2337]: 找 guanhuanping 确认下 注释 Formula，是否需要

Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2337]: 已修改

**Review by wushaoqiang2168 — APPROVED** [review]

File: bangc-ops/mlu_op.h [hunk 2]
```diff
@@ -2293,6 +2293,240 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+     output array by 1 * 2 * 2 * 1 -->
+         output: [[[[1],[1]],[[1],[1]]]]
+     @endverbatim
+ *
+ * @par Reference
+ * - github.com/open-mmlab/mmcv/blob/master/mmcv/ops/roi_align_rotated.py
+ */
+mluOpStatus_t MLUOP_WIN_API
+mluOpRoiAlignRotatedForward(mluOpHandle_t handle,
+                           const mluOpTensorDescriptor_t features_desc,
```
Comment (WangYuan-cn) [inline bangc-ops/mlu_op.h:2400]: 没对齐

Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2400]: 已修改

File: bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+#include "core/context.h"
+#include "core/gen_case.h"
+#include "core/logging.h"
+#include "core/runtime/device.h"
+#include "core/tensor.h"
+#include "core/type.h"
+#include "mlu_op.h"
+#include "mlu_op_kernel.h"
+
+static void policyFunc(mluOpHandle_t handle, int bin_num, cnrtDim3_t *k_dim,
```
Comment (mahxn0) [inline bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp:34]: const

File: bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+                       cnrtFunctionType_t *k_type) {
+  unsigned int core_num = handle->core_num_per_cluster;
+  unsigned int cluster_num =
+      mluop::runtime::getJobLimitCapability(handle) / core_num;
+  *k_type = CNRT_FUNC_TYPE_UNION1;
+  k_dim->x = core_num;
+  unsigned int use_cluster = (bin_num + core_num - 1) / core_num;
```
Comment (mahxn0) [inline bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp:41]: bin_num可能大于2g  size_t use_cluster

File: bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+               << ", output_desc channel = " << output_desc->dims[3]
+               << ". They should be the same.";
+    return MLUOP_STATUS_BAD_PARAM;
+  }
+
+  int channel = features_desc->dims[3];
+  int width = features_desc->dims[2];
+  int height = features_desc->dims[1];
+  int batch = features_desc->dims[0];
+  int rois_nums = rois_desc->dims[0];
```
Comment (mahxn0) [inline bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp:96]: const

File: bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_align_rotated_backward/roi_align_rotated_backward.cpp [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+
+void RoiAlignRotatedBackwardExecutor::preCalcForBilinearInterpolate(
+    const int height, const int width, const int channel,
+    const int pooled_height, const int pooled_width, const int roi_bin_grid_h,
+    const int roi_bin_grid_w, const float roi_start_x, const float roi_start_y,
+    const float bin_size_h, const float bin_size_w, const float roi_center_x,
+    const float roi_center_y, const float cos_theta, const float sin_theta,
+    std::vector<PreCalc> &pre_calc) {
+  int pre_calc_idx = 0;
+  for (int ph = 0; ph < pooled_height; ++ph) {
```
Comment (mahxn0) [inline bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_align_rotated_backward/roi_align_rotated_backward.cpp:38]: 对齐了吗这里

**Review by mahxn0 — APPROVED** [review]

File: bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp (hunk 5 above)
Comment (tudejiang79) [reply bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp:96]: 已修改

File: bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_align_rotated_backward/roi_align_rotated_backward.cpp (hunk 6 above)
Comment (tudejiang79) [reply bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_align_rotated_backward/roi_align_rotated_backward.cpp:38]: 已修改

File: bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp (hunk 4 above)
Comment (tudejiang79) [reply bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp:41]: 已修改

File: bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp (hunk 3 above)
Comment (tudejiang79) [reply bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp:34]: 已修改

File: docs/bangc-docs/user_guide/9_operators/index.rst [hunk 7]
```diff
@@ -251,3 +251,17 @@ mluOpPsamaskForward
...
+
+其中 rois 是一个二维的Tensor，其第一维度与 output 的第一维度相同，最后一维必须等于 6 。每个 roi 包含（batch_id, x, y, w, h, theta），其中，x 和 y 表示的是 roi 中心点的坐标，w 和 h 分别是 roi 的宽和高，theta 表示边框逆时针旋转的角度。
+
+rois 中 batch_id 的值在 [0, batch-1] 范围内，其中 batch 是输入 featrues 的第一维的大小。
+
+output 的最高维与 rois 的最高维度相等，最后一维度大小与 features 的最后一维相等。
+
+mluOpRoiAlignRotatedBackward
+-------------------
+mluOpRoiAlignRotatedForward 算子的反向
```
[the remark this answers is not in the document]
Comment (tudejiang79) [reply docs/bangc-docs/user_guide/9_operators/index.rst:267]: 已修改

File: bangc-ops/mlu_op.h [hunk 8]
```diff
@@ -2293,6 +2293,240 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+     param:
+            pooled_height: 2, pooled_width: 2, spatial_scale: 1.0,
+            sampling_ratio: 2, aligned: false, clockwise: false
+
+     output array by 1 * 2 * 2 * 1 -->
+         output: [[[[1],[1]],[[1],[1]]]]
+     @endverbatim
+ *
+ * @par Reference
+ * - github.com/open-mmlab/mmcv/blob/master/mmcv/ops/roi_align_rotated.py
```
Comment (WangYuan-cn) [inline bangc-ops/mlu_op.h:2396]: 1

Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2396]: 已修改

File: bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+                             spatial_scale);
+    GEN_CASE_OP_PARAM_SINGLE(1, "roi_align_rotated_forward", "aligned",
+                             aligned);
+    GEN_CASE_OP_PARAM_SINGLE(2, "roi_align_rotated_forward", "clockwise",
+                             clockwise);
+    GEN_CASE_TEST_PARAM_NEW(true, true, false, 0.003, 0.003, 0);
+  }
+  mluOpRoiAlignRotatedParams roiAlignRotatedParams{pooled_height, pooled_width,
+                                                   sample_ratio,  spatial_scale,
+                                                   aligned,       clockwise};
```
Comment (WangYuan-cn) [inline bangc-ops/kernels/roi_align_rotated/roi_align_rotated.cpp:139]: 这里对齐是刷出来的么

**Review by WangYuan-cn — APPROVED** [review]

File: bangc-ops/mlu_op.h [hunk 10]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
                        const mluOpTensorDescriptor_t bottom_grad_desc,
                        void *bottom_grad);
 
+// Group:RoiAlignRotated
+/*!
+ * @brief According to the \b rois with rotation, extracts the corresponding \b features information to \b output
+ * by bilinear interpolation.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2299]: Extracts the corresponding \b features information to \b output by bilinear interpolation according to the \b rois with rotation.   调整一下顺序，把核心描述放到前面。

File: bangc-ops/mlu_op.h [hunk 11]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ * @param[in] handle
+ * Handle to a Cambricon MLUOP context that is used to manage MLU devices and queues in
+ * ::mluOpRoiAlignRotatedForward operation. For detailed information, see ::mluOpHandle_t.
+ * @param[in] features_desc
+ * The descriptor of the features tensor.
+ * @param[in] features
+ * Pointer to the MLU memory that stores the features tensor. The shape of \b features 
+ * is [batch_num, H, W, C].
+ * @param[in] rois_desc
+ * Descriptor of rois tensor, containing dimension and the layout of rois.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2310]: The descriptor of rois tensor, which contains dimension and the layout of rois.

File: bangc-ops/mlu_op.h [hunk 12]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ * @param[in] rois
+ * Pointer to the MLU memory that stores rois tensors. \b rois[i] consists of [batch_id, 
+ * x, y, w, h, theta], where \p batch_id is the ID of the batch, \p x and \p y are the coordinate
+ * of center point, \p w and \p h are the width and height of rois, and \p theta is the rotated angle.
+ * @param[in] pooled_height
+ * The height of output.
+ * @param[in] pooled_width
+ * The width of output.
+ * @param[in] sample_ratio
+ * The number of sampling points in the bin used to compute the output.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2321]: The number of sampling points in the bin which is used to compute the output.

File: bangc-ops/mlu_op.h (hunk 10 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2299]: 已修改

File: bangc-ops/mlu_op.h (hunk 11 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2310]: 已修改

File: bangc-ops/mlu_op.h [hunk 13]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ * @param[in] spatial_scale
+ * The spatial scale of each regions of interest in the output.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2323]: regions of interest > ROI

File: bangc-ops/mlu_op.h [hunk 14]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ * @param[in] aligned
+ * A boolean value which determines whether to shift the roi by 0.5 pixel. If the
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2325]: roi 》ROI

File: bangc-ops/mlu_op.h [hunk 15]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ * @param[in] pooled_width
+ * The width of output.
+ * @param[in] sample_ratio
+ * The number of sampling points in the bin used to compute the output.
+ * @param[in] spatial_scale
+ * The spatial scale of each regions of interest in the output.
+ * @param[in] aligned
+ * A boolean value which determines whether to shift the roi by 0.5 pixel. If the 
+ * value of \b aligned is set to true, the roi is shifted by 0.5. If the value of \b aligned 
+ * is set to false, the roi is not shifted.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2327]: roi全部大写

File: bangc-ops/mlu_op.h (hunk 12 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2321]: 已修改

File: bangc-ops/mlu_op.h (hunk 13 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2323]: 已修改

File: bangc-ops/mlu_op.h [hunk 16]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ * @param[in] clockwise
+ * A boolean value which determines whether the rotation of roi is clockwise.
+ * @param[out] output_desc
+ * Descriptor of output, containing dimension and the layout of output.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2331]: The descriptor of output, which contains dimension and the layout of output.

File: bangc-ops/mlu_op.h [hunk 17]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ *
+ * @par Data Type
+ * - This function supports the following data types for input tensor \b features, \b rois,
+ *   and output tensor \b output. Data type of all tensors should be the same.
+ *   - input tensor: half, float.
+ *   - rois tensor: half, float.
+ *   - output tensor: half, float.
+ *
+ * @par Data Layout
+ * - The supported data layout of \b features, \b rois, and \b output are as follows:
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2346]: layouts

File: bangc-ops/mlu_op.h (hunk 14 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2325]: 已修改

File: bangc-ops/mlu_op.h [hunk 18]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ *   - output tensor: \p MLUOP_LAYOUT_NHWC.
+ *
+ * @par Scale Limitation
+ * - The \b features tensor and \b output tensor should be 4D.
+ * - The half data type is not recommended due to low precision.
+ * - Size of the lowest dimension of \b features tensor and \b output tensor should be the same.
+ * - The \b rois tensor should be 2D array.
+ * - Size of the highest dimension of \b output tensor and \b rois tensor should be the same.
+ * - The shape of \b rois should be [rois_num, 6].
+ * - \p batch_id should be in the range of [0, \p batch_num - 1], \p x and \p y should be greater than or
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2358]: 逗号改为英文分号;

File: bangc-ops/mlu_op.h (hunk 15 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2327]: 已修改

File: bangc-ops/mlu_op.h (hunk 16 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2331]: 已修改

File: bangc-ops/mlu_op.h (hunk 17 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2346]: 已修改

File: bangc-ops/mlu_op.h [hunk 19]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+                            const float spatial_scale,
+                            const bool aligned,
+                            const bool clockwise,
+                            const mluOpTensorDescriptor_t output_desc,
+                            void* output);
+
+// Group:RoiAlignRotated
+/*!
+ * @brief Computes the gradients of feature map \b bottom_grad based on the input \b top_grad and
+ * \b rois to perform the backpropagation of the ::mluOpRoiAlignRotatedForward operator.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2408]: 通篇算子的英文，使用operation或operator都可以，保持一致吧

File: bangc-ops/mlu_op.h [hunk 20]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ *
+ * @param[in] handle
+ * Handle to a Cambricon MLUOP context that is used to manage MLU devices and queues in
+ * ::mluOpRoiAlignRotatedBackward operation. For detailed information, see ::mluOpHandle_t.
+ * @param[in] top_grad_desc
+ * The descriptor of the gradient tensor in the backpropagation process.
+ * @param[in] top_grad
+ * Pointer to the MLU memory that stores the top_grad tensor.
+ * @param[in] rois_desc
+ * Descriptor of rois tensor, containing dimension and the layout of rois.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2418]: The descriptor of rois tensor, which contains dimension and the layout of rois.

File: bangc-ops/mlu_op.h (hunk 18 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2358]: 已修改

File: bangc-ops/mlu_op.h [hunk 21]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ * and \p y are the coordinate of center point, \p w and \p h are the width and height
+ * of rois, and \p theta is the rotated angle.
+ * @param[in] pooled_height
+ * The height of output.
+ * @param[in] pooled_width
+ * The width of output.
+ * @param[in] sample_ratio
+ * The number of sampling points in the bin used to compute the output.
+ * @param[in] spatial_scale
+ * The spatial scale of each regions of interest in the output.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2432]: regions of interest > ROI

File: bangc-ops/mlu_op.h (hunk 19 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2408]: 已修改

File: bangc-ops/mlu_op.h [hunk 22]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ * @param[in] aligned
+ * A boolean value which determines whether to shift the roi by 0.5 pixel. 
+ * If the value of \b aligned is set to true, the roi is shifted by 0.5. If the value
+ * of \b aligned is set to false, the roi is not shifted.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2436]: roi改大写

File: bangc-ops/mlu_op.h [hunk 23]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ * @param[out] bottom_grad
+ * Pointer to the MLU memory that stores the bottom_grad tensor. The shape of 
+ * bottom_grad is [batch_num, H, W, C].
+ * 
+ * @par Return
+ * - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM.
+ *
+ * @par Data Type
+ * - This function supports the following data types for input tensor \b top_grad, \b rois,
+ *   and output tensor \b bottom_grad. Data type of all tensors should be the same.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2450]: Data types

File: bangc-ops/mlu_op.h [hunk 24]
```diff
@@ -2293,6 +2293,224 @@ mluOpPsRoiPoolBackward(mluOpHandle_t handle,
...
+ *   - top_grad tensor: half, float.
+ *   - rois tensor: half, float.
+ *   - bottom_grad tensor: half, float.
+ *
+ * @par Data Layout
+ * - The supported data layout of \b top_grad, \b rois, and \b bottom_grad are as follows:
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2456]: layouts

File: bangc-ops/mlu_op.h (hunk 21 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2432]: 已修改

File: bangc-ops/mlu_op.h (hunk 22 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2436]: 已修改

**Review by GraceGuanH — COMMENTED** [review]
please check the comments and amend them.

File: bangc-ops/mlu_op.h (hunk 23 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2450]: 已修改

File: bangc-ops/mlu_op.h (hunk 24 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2456]: 已修改

**Review by GraceGuanH — APPROVED** [review]

**Pull request merged**